### PR TITLE
HPCC-28881 Extend configuration update hook mechanism for bare-metal

### PR DESCRIPTION
**Pull request by jakesmith**
Allow existing config update mechanism to work in bare-metal too, and extent it to also spot Dali Environment updates.

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: system/jlib/jptree.cpp [hunk 1]
```diff
@@ -9014,16 +9024,12 @@ jlib_decl IPropertyTree * loadConfiguration(IPropertyTree *componentDefault, con
      * on-demand, which means this mechanism is sufficient to cover most cases.
      */
 
-#if defined(__linux__) && defined(_CONTAINERIZED)
+#if defined(__linux__)
     /* In bare-metal - there is no auto process/component restart mechanism, so everything would need to be
```
Comment (mckellyln) [inline system/jlib/jptree.cpp:9028]: Maybe remove comment now ?

Comment (jakesmith) [reply system/jlib/jptree.cpp:9028]: yes, and/or comment above it needs updaing. I'll do that.

File: dali/base/daclient.cpp [hunk 2]
```diff
@@ -102,6 +140,25 @@ bool initClientProcess(IGroup *servergrp, DaliClientRole role, unsigned mpport,
...
+        case DCR_ThorMaster:
+        case DCR_EclServer:
+        case DCR_EclAgent:
+        case DCR_SashaServer:
+        case DCR_DfuServer:
+        case DCR_EspServer:
+        case DCR_Roxie:
+        case DCR_AgentExec:
+        case DCR_EclScheduler:
+        case DCR_EclCCServer:
```
Comment (mckellyln) [inline dali/base/daclient.cpp:157]: DCR_DaliServer ?

Comment (mckellyln) [reply dali/base/daclient.cpp:157]: ok, so only clients

Comment (jakesmith) [reply dali/base/daclient.cpp:157]: yes, but some cli tools and legacy tools etc. are also excluded from this monitoring process

**Comment by jakesmith** [comment]
@mckellyln - please see updated comments in jptree.cpp in new commit

**Review by mckellyln — APPROVED** [review]
Approved

**Comment by jakesmith** [comment]
Squashed

**Review by ghalliday — APPROVED** [review]
@jakesmith please squash

**Pull request merged**